mobileapp/components/game/Game.test.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react-native";
+import { Game } from "./Game";
+
+jest.mock("@react-navigation/native", () => {
+  const actualNav = jest.requireActual("@react-navigation/native");
+  return {
+    ...actualNav,
+    useRoute: () => {
+      params: {
+        itemId: "1";
+      }
+    },
+  };
+});
+
+describe("Game", () => {
+  it("displays loader first", async () => {
+    const renderedGame = render(<Game />);
+    const loader = renderedGame.getByTestId("loader");
+
+    expect(loader).toBeDefined();
+  });
+
+  it("displays refresh button", async () => {
+    render(<Game />);
+    const refreshButton = await screen.findByTestId("refreshButton");
```

Comment:
You use 2 ways to get an element, `renderedGame.getByTestId` and `screen.findByTestId` (why `findByTestId` instead of `getByTestId` by the way?), you should use only one. Plus the best practice is to use `screen` (https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen).

Also, another best practice is to use `getByRole` as much as you can, this ensure that the element is accessible not for the test, but in general. I personnaly rarely use `testIds`. 
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query
https://testing-library.com/docs/queries/about/#priority

Reply:
I used findByTestId and getByTestId because I both needed to check first render (for the loader) and render after a failed data fetching (for refresh button).

I'll try and use getByRole instead of getByTestId.